realsense_camera/src/base_nodelet.cpp

```diff
@@ -246,23 +246,30 @@ namespace realsense_camera
    */
   void BaseNodelet::advertiseTopics()
   {
-    image_transport::ImageTransport image_transport(nh_);
-    camera_publisher_[RS_STREAM_COLOR] = image_transport.advertiseCamera(COLOR_TOPIC, 1);
-    camera_publisher_[RS_STREAM_DEPTH] = image_transport.advertiseCamera(DEPTH_TOPIC, 1);
-    camera_publisher_[RS_STREAM_INFRARED] = image_transport.advertiseCamera(IR_TOPIC, 1);
-    pointcloud_publisher_ = nh_.advertise<sensor_msgs::PointCloud2>(PC_TOPIC, 1);
+    ros::NodeHandle color_nh(nh_, COLOR_NAMESPACE);
+    image_transport::ImageTransport color_image_transport(color_nh);
+    camera_publisher_[RS_STREAM_COLOR] = color_image_transport.advertiseCamera(COLOR_TOPIC, 1);
+
+    ros::NodeHandle depth_nh(nh_, DEPTH_NAMESPACE);
+    image_transport::ImageTransport depth_image_transport(depth_nh);
+    camera_publisher_[RS_STREAM_DEPTH] = depth_image_transport.advertiseCamera(DEPTH_TOPIC, 1);
+    pointcloud_publisher_ = depth_nh.advertise<sensor_msgs::PointCloud2>(PC_TOPIC, 1);
+
+    ros::NodeHandle ir_nh(nh_, IR_NAMESPACE);
+    image_transport::ImageTransport ir_image_transport(ir_nh);
+    camera_publisher_[RS_STREAM_INFRARED] = ir_image_transport.advertiseCamera(IR_TOPIC, 1);
   }
 
   /*
    * Advertise services.
    */
   void BaseNodelet::advertiseServices()
   {
-    get_options_service_ = nh_.advertiseService(SETTINGS_SERVICE, &BaseNodelet::getCameraOptionValues, this);
-    set_power_service_ = nh_.advertiseService(CAMERA_SET_POWER_SERVICE, &BaseNodelet::setPowerCameraService, this);
-    force_power_service_ = nh_.advertiseService(CAMERA_FORCE_POWER_SERVICE, &BaseNodelet::forcePowerCameraService, this);
-    is_powered_service_ = nh_.advertiseService(CAMERA_IS_POWERED_SERVICE, &BaseNodelet::isPoweredCameraService, this);
-}
+    get_options_service_ = pnh_.advertiseService(SETTINGS_SERVICE, &BaseNodelet::getCameraOptionValues, this);
+    set_power_service_ = pnh_.advertiseService(CAMERA_SET_POWER_SERVICE, &BaseNodelet::setPowerCameraService, this);
```

Comment:
@tfoote Is it OK to have a standard where users are expected to make the topics, services and parameters of a nodelet instance unique, by grouping them using "**group ns**"; thereby avoiding the usage of **PrivateNodeHandle** in the code and also avoiding to usage of the nodelet name to access them.

E.g. **group ns = "camera"** will make the **topic as "camera/depth/image_raw"**, the **service as "camera/get_settings"** and **parameter as "camera/mode"**.